okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -155,14 +156,30 @@ private static ResponseBody buffer(MediaType mediaType, byte[] bodyBytes) {
 
     @Override
     public void enqueue(Callback callback) {
-        AsyncTracer tracer = new AsyncTracer("OkHttp: acquire-limiter");
+        AsyncTracer tracer = new AsyncTracer("OkHttp: acquire-limiter"); // start 'acquire-limiter-enqueue'
+
         ListenableFuture<Limiter.Listener> limiterListener = limiter.acquire();
         request().tag(ConcurrencyLimiterListener.class).setLimiterListener(limiterListener);
+
         Futures.addCallback(limiterListener, new FutureCallback<Limiter.Listener>() {
             @Override
             public void onSuccess(Limiter.Listener listener) {
-                tracer.withTrace(() -> null);
-                enqueueInternal(callback);
+                // tracer.withTrace completes the 'acquire-limiter-enqueue'
+                // tracer.withTrace starts the 'acquire-limiter-run' span
+                tracer.withTrace(() -> {
```

Comment:
I think before merging this it would be nice to come up with a solution that doesn't require us to include the 'acquire-limiter-run` and `ignore` span hacks

Reply:
That was my thought as well - this pr is here to point to intended behaviour

Reply:
https://github.com/palantir/tracing-java/pull/124

Reply:
Why was the hack here needed?
It seems like leaving the old behavior would suffice to leave the `acquire-limiter-run` span empty and the other spans with the appropriate parents:
```
tracer.withTrace(() -> null);
enqueueInternal(callback);
```

Reply:
It won't work - the callback is executed on different thread and different trace. You need to add spans to the original trace